src/many-sources/abstract-pool.js

```diff
@@ -37,12 +37,10 @@ inherit(AbstractPool, Stream, {
       } else if (this._drop === 'old') {
         if (this._overlapping) {
           this._addToCur(toObs(obj));
-          while (this._curSources.length > this._concurLim) {
-            this._removeOldest();
-          }
+          this._removeOldest();
         } else {
           this._removeOldest();
-          this._add(obj, toObs);
+          this._addToCur(toObs(obj));
```

Comment:
while cleaner; this doesn't _feel_ right. see my previous comment, I need to think more about this/this is idea is more subtle than I initially projected.